src/moves.rs

```diff
@@ -3,6 +3,26 @@ use super::hand::Hands;
 use super::judge::{Judge, Outcome};
 use crate::error::GamePlayError;
 
+#[derive(Debug, PartialEq)]
+pub enum ChangeAction {
+    Added,
+    Swapped,
```

Comment:
I was imagining that `ChangeAction`s would define the relevant details. I.e., `Swapped(ChangeDetail, ChangeDetail)`. Right now if you have 4 `Swapped` changes you can't tell how they pair up.

The current way makes sense if we're just imagining highlighting squares based on their `ChangeAction`. But another nice use would be an `explain` method on each change that could say "The B at (0,1) was swapped with the A at (1,0)", for example. That would require putting details on the actions, I think.

Reply:
> I was imagining that ChangeActions would define the relevant details. I.e., Swapped(ChangeDetail, ChangeDetail). Right now if you have 4 Swapped changes you can't tell how they pair up.

The reason I didn't is that the change array contains only the changes out of a single turn, and you can't swap 4 tiles in a certain turn, so in our current ruleset the Swapped action is fully described

Reply:
Future goal with nice explanations sounds good though :) But I don't know whether that needs to be done by the game logic itself or if that high-level detail can be synthesized by our UI